inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/CompressFunction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import lombok.extern.slf4j.Slf4j;
+import net.sf.jsqlparser.expression.Function;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.zip.Deflater;
+
+/**
+ * CompressFunction
+ * description: compress(string_to_compress) : Compresses a string
+ * - Return NULL if string_to_compress is NULL.
+ * - Return "" if string_to_compress is "".
+ * - Return the result as a binary string.
+ */
+@Slf4j
+@TransformFunction(names = {"compress"})
+public class CompressFunction implements ValueParser {
+
+    private final ValueParser stringParser;
+    // ISO-8859-1 encoding does not convert negative numbers and can preserve the original values of byte arrays.
+    private final Charset CHARSET = StandardCharsets.ISO_8859_1;
+
+    public CompressFunction(Function expr) {
+        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
```

Comment:
Miss the parameter of compress type

Reply:
As stated on the MySQL official website（ https://dev.mysql.com/doc/refman/8.4/en/encryption-functions.html#function_compress ）, Compress Function uses zlib for string compression, and only needs to provide the string to be compressed as the parameter of the function. The CHARSET here is used to perform lossless conversion between the compressed byte array and string to meet the output requirements of Compress().

Reply:
provide a two arguments method like compress(string_to_compress, compress_type)

Reply:
done.